### PR TITLE
feat(OscVm): add userdata on nodes for tags usage

### LINKED ISSUE #331: [Feature]: add userdata and tags on OscVm
### Explain problem to solve

We need to use the userdata with tag `osc.fcu.repulse_server` on our nodes as describe [here](https://docs.outscale.com/fr/userguide/Configurer-une-VM-avec-les-user-data-et-les-tags-OUTSCALE.html)

### Describe the solution you would like

add a userdata field on the OscVm object

### Additional context

-

### Environment

```markdown
- Kubernetes version: (use `kubectl version`): -
- OS (e.g. from `/etc/os-release`):
- Kernel (e.g. `uname -a`):
- cluster-api-provider-outscale version:
- cluster-api version:
- Install tools:
- Kubernetes Distribution:
- Kubernetes Diestribution version:
```

### PR DESCRIPTION
**Pull request by sebglon**
**What type of PR is this?**

/kind feature           Add userdata and tags on OscVm.

**What this PR does / why we need it**:
We need it to add tags on VM as describe [here](https://docs.outscale.com/fr/userguide/Configurer-une-VM-avec-les-user-data-et-les-tags-OUTSCALE.html)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #331

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests

### CODE REVIEW

File: example/cluster-machine-template-multi-az.yaml [hunk 1]
```diff
@@ -151,14 +151,14 @@ spec:
       node:
         clusterName: cluster-api-multi-az
         image:
-         name: ubuntu-2004-2004-kubernetes-v1.22.11-2022-08-22
+         name: ubuntu-2004-2004-kubernetes-v1.27.7-2023-10-18
         keypair:
-          name: cluster-api
+          name: sebastien
```
Comment (outscale-hmi) [inline example/cluster-machine-template-multi-az.yaml:156]: Can you please update the name

File: example/cluster-machine-template-multi-az.yaml [hunk 2]
```diff
@@ -151,14 +151,14 @@ spec:
...
           deleteKeypair: false
         vm:
           clusterName: cluster-api-multi-az
           name: cluster-api-multi-az-vm-control-plane
-          keypairName: cluster-api
+          keypairName: sebastien
```
Comment (outscale-hmi) [inline example/cluster-machine-template-multi-az.yaml:161]: idem

File: example/cluster-machine-template-multi-az.yaml [hunk 3]
```diff
@@ -223,14 +225,14 @@ spec:
       node:
         clusterName: cluster-api-multi-az
         image:
-         name: ubuntu-2004-2004-kubernetes-v1.22.11-2022-08-22
+         name: ubuntu-2004-2004-kubernetes-v1.27.7-2023-10-18
         keypair:
-          name: cluster-api-test
+          name: sebastien
```
Comment (outscale-hmi) [inline example/cluster-machine-template-multi-az.yaml:230]: idem

File: example/cluster-machine-template-multi-az.yaml [hunk 4]
```diff
@@ -223,14 +225,14 @@ spec:
...
           deleteKeypair: false
         vm:
           clusterName: cluster-api-multi-az
           name: cluster-api-multi-az-vm-worker-2a
-          keypairName: cluster-api
+          keypairName: sebastien
```
Comment (outscale-hmi) [inline example/cluster-machine-template-multi-az.yaml:235]: idem

File: example/cluster-machine-template-multi-az.yaml [hunk 5]
```diff
@@ -253,14 +255,14 @@ spec:
       node:
         clusterName: cluster-api-multi-az
         image:
-         name: ubuntu-2004-2004-kubernetes-v1.22.11-2022-08-22
+         name: ubuntu-2004-2004-kubernetes-v1.27.7-2023-10-18
         keypair:
-          name: cluster-api-test
+          name: sebastien
```
Comment (outscale-hmi) [inline example/cluster-machine-template-multi-az.yaml:260]: idem

File: example/cluster-machine-template-multi-az.yaml [hunk 6]
```diff
@@ -253,14 +255,14 @@ spec:
...
           deleteKeypair: false
         vm:
           clusterName: cluster-api-multi-az
           name: cluster-api-multi-az-vm-worker-2b
-          keypairName: cluster-api
+          keypairName: sebastien
```
Comment (outscale-hmi) [inline example/cluster-machine-template-multi-az.yaml:265]: idem

File: hack/ensure-clusterctl.sh [hunk 7]
```diff
@@ -45,7 +45,7 @@ EOF
 }
 
 install_clusterctl() {
-    if [[ "${OSTYPE}" == "linux"* || "${OSTYPE}" == "darwin22"* ]]; then
+    if [[ "${OSTYPE}" == "linux"* || "${OSTYPE}" == "darwin"* ]]; then
```
Comment (outscale-hmi) [inline hack/ensure-clusterctl.sh:48]: oui. this change, is not rellated to this PR, can you please push it in another PR, as this is not a trivial changes.

File: example/cluster-machine-template-simple.yaml [hunk 8]
```diff
@@ -300,14 +302,14 @@ spec:
       node:
         clusterName: cluster-api
         image:
-         name: ubuntu-2004-2004-kubernetes-v1.22.11-2022-08-22
+         name: ubuntu-22.04-2204-kubernetes-v1.29.1-2024-02-19
         keypair:
-          name: cluster-api
+          name: sebastien
```
Comment (outscale-hmi) [inline example/cluster-machine-template-simple.yaml:307]: idem

File: example/cluster-machine-template-simple.yaml [hunk 9]
```diff
@@ -300,14 +302,14 @@ spec:
...
           deleteKeypair: false
         vm:
           clusterName: cluster-api
           name: cluster-api-vm-kcp
-          keypairName: cluster-api
+          keypairName: sebastien
```
Comment (outscale-hmi) [inline example/cluster-machine-template-simple.yaml:312]: idem

**Review by outscale-hmi — COMMENTED** [review]
Good job,

File: hack/ensure-clusterctl.sh (hunk 7 above)
Comment (sebglon) [reply hack/ensure-clusterctl.sh:48]: Done here https://github.com/outscale/cluster-api-provider-outscale/pull/336

**Review by outscale-hmi — APPROVED** [review]
ok

**Comment by outscale-hmi** [comment]
@sebglon  can you check unitTest  and verify-boilerplate please ?

**Comment by sebglon** [comment]
@outscale-hmi i can't run e2e on my side. it's not easy to debug.
I push some fix but not confident.

**Comment by sebglon** [comment]
@outscale-hmi can i have a review please?

**Pull request merged**